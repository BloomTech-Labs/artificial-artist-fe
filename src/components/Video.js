import React, { useState } from "react";
import "../../node_modules/video-react/dist/video-react.css";
// import { Player, BigPlayButton } from "video-react";
import style from "styled-components";
import ReactPlayer from "react-player/lazy";

const VideoContainer = style.div`
  // width: ${(props) => (props.heroVideo ? "40%" : "100%")};
  width: 100%;
  height: calc(100% - 2px);
  margin: 0 auto;
  display: block;
  position: relative;
`;

const VideoPlayPause = style.img`
  top: calc(50% - 35px);
  left: calc(50% - 35px);
  position: absolute;
  cursor: pointer;
  display: ${(props) => (props.playing ? "none" : "block")};
  transition: all .25s ease-in-out;
  &:hover {
    transform: scale(1.2);
  }
`;

const ReactPlayerExt = style(ReactPlayer)`
  height: calc(100% + 4px) !important;
`;

const Video = (props) => {
  const [playerState, setPlayerState] = useState({
    playing: false,
    controls: false,
  });

  const handlePlayPause = () => {
    setPlayerState({ ...playerState, playing: !playerState.playing });
  };

  return (
    <>
      <VideoContainer heroVideo={props.heroVideo} onClick={handlePlayPause}>
        <ReactPlayerExt
          playsinline
          config={{ file: { attributes: { poster: props.video.thumbnail } } }}
          playing={playerState.playing}
          controls={playerState.controls}
          width="100%"
          height="100%"
          url={
            props.video
              ? props.video.location
              : `${process.env.REACT_APP_S3VIDEOS}billieeilish--you_should_see_me_in_a_crown-king_me_-coLerbRvgsQ.mp4`
          }
        ></ReactPlayerExt>
        <VideoPlayPause
          playing={playerState.playing}
          src="../images/icon-video-play.svg"
        />
      </VideoContainer>
    </>
  );
};

export default Video;
