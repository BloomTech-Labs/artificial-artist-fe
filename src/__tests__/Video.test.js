import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Video from "../components/Video";

describe("VideoPlayer", () => {
  const video = {
    video_title: "test",
    location: "test",
    thumbnail: "test.png",
    video_status: "successful",
    user_id: 1,
    song_id: 1,
  };

  test("VideoPlayer renders", () => {
    shallow(<Video video={video} />);
  });

  test("should have video player element", () => {
    const wrapper = shallow(<Video video={video} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
