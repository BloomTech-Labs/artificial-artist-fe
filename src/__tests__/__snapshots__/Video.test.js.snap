// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer should have video player element 1`] = `
<Fragment>
  <styled.div
    onClick={[Function]}
  >
    <Styled(ReactPlayer)
      config={
        Object {
          "file": Object {
            "attributes": Object {
              "poster": "test.png",
            },
          },
        }
      }
      controls={false}
      height="100%"
      playing={false}
      playsinline={true}
      url="test"
      width="100%"
    />
    <styled.img
      playing={false}
      src="../images/icon-video-play.svg"
    />
  </styled.div>
</Fragment>
`;
